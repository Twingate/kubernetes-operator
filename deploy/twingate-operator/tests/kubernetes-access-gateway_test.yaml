suite: Kubernetes Access Gateway
templates:
  - charts/kubernetes-access-gateway/templates/*.yaml
set:
  twingateOperator:
    network: <network slug>
    apiKey: <api key>
    remoteNetworkId: <remote network id>
tests:
  - it: should enable Kubernetes Access Gateway
    # Hardcode the chart version and app version to ensure that the snapshot is stable
    # when upgrading Kubernetes Access Gateway subchart
    chart:
      version: 1.0.0
      appVersion: 1.0.0
    set:
      kubernetes-access-gateway:
        enabled: true
        twingate:
          network: <network slug>
        tls:
          autoGenerated: false
          cert: <tls cert>
          key: <tls key>
          ca: <ca cert>
    asserts:
      - matchSnapshot: {}
