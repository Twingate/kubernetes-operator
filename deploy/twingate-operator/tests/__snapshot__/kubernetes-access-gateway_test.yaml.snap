should enable Kubernetes Access Gateway:
  1: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRole
    metadata:
      name: RELEASE-NAME-kubernetes-access-gateway-cluster-role
    rules:
      - apiGroups:
          - ""
        resources:
          - users
          - groups
          - serviceaccounts
        verbs:
          - impersonate
  2: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRoleBinding
    metadata:
      name: RELEASE-NAME-kubernetes-access-gateway-cluster-role-binding
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: ClusterRole
      name: RELEASE-NAME-kubernetes-access-gateway-cluster-role
    subjects:
      - kind: ServiceAccount
        name: RELEASE-NAME-kubernetes-access-gateway
        namespace: NAMESPACE
  3: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: kubernetes-access-gateway
        app.kubernetes.io/version: 1.0.0
        helm.sh/chart: kubernetes-access-gateway-1.0.0
      name: RELEASE-NAME-kubernetes-access-gateway
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: kubernetes-access-gateway
      template:
        metadata:
          annotations:
            checksum/tlsAlternativeNames: eee78e4712e57977935ceb9ebfdb354aa33c85ca4dbb490001827f64259b6b8d
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: kubernetes-access-gateway
            app.kubernetes.io/version: 1.0.0
            helm.sh/chart: kubernetes-access-gateway-1.0.0
        spec:
          containers:
            - env:
                - name: TWINGATE_NETWORK
                  value: <network slug>
                - name: TWINGATE_HOST
                  value: twingate.com
                - name: TWINGATE_PORT
                  value: "8443"
                - name: TWINGATE_METRICSPORT
                  value: "9090"
              image: twingate/kubernetes-access-gateway:1.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: https
                  scheme: HTTPS
                initialDelaySeconds: 5
                periodSeconds: 5
              name: kubernetes-access-gateway
              ports:
                - containerPort: 8443
                  name: https
                  protocol: TCP
                - containerPort: 9090
                  name: metrics
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /healthz
                  port: https
                  scheme: HTTPS
                initialDelaySeconds: 5
                periodSeconds: 5
              resources: {}
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
                readOnlyRootFilesystem: true
                runAsNonRoot: true
                runAsUser: 1000
              volumeMounts:
                - mountPath: /etc/tls-secret-volume
                  name: tls-secret-volume
                  readOnly: true
          securityContext:
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: RELEASE-NAME-kubernetes-access-gateway
          volumes:
            - name: tls-secret-volume
              secret:
                secretName: RELEASE-NAME-kubernetes-access-gateway-tls
  4: |
    apiVersion: v1
    kind: Service
    metadata:
      annotations: null
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: kubernetes-access-gateway
        app.kubernetes.io/version: 1.0.0
        helm.sh/chart: kubernetes-access-gateway-1.0.0
      name: RELEASE-NAME-kubernetes-access-gateway
    spec:
      ports:
        - name: https
          port: 443
          protocol: TCP
          targetPort: https
      selector:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: kubernetes-access-gateway
      type: ClusterIP
  5: |
    apiVersion: v1
    kind: ServiceAccount
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: kubernetes-access-gateway
        app.kubernetes.io/version: 1.0.0
        helm.sh/chart: kubernetes-access-gateway-1.0.0
      name: RELEASE-NAME-kubernetes-access-gateway
  6: |
    apiVersion: v1
    data:
      ca.crt: PGNhIGNlcnQ+
      tls.crt: PHRscyBjZXJ0Pg==
      tls.key: PHRscyBrZXk+
    kind: Secret
    metadata:
      name: RELEASE-NAME-kubernetes-access-gateway-tls
    type: kubernetes.io/tls
